### PR TITLE
Support getting RELR relocations from dynamic section

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

**Comment by sevaa** [comment]
A binary to test that against would be  nice.

File: elftools/elf/relocation.py [hunk 1]
```diff
@@ -182,6 +193,16 @@ def get_relocation(self, n):
             self._cached_relocations = list(self.iter_relocations())
         return self._cached_relocations[n]
 
+
+class RelrRelocationSection(Section, RelrRelocationTable):
```
Comment (eliben) [inline elftools/elf/relocation.py:197]: I'd rather use composition than multiple inheritance, unless there's a strong reason to do the latter.

**Comment by eliben** [comment]
@sevaa other than the test comment, LGTY?

**Comment by sevaa** [comment]
Never really understood how do relocations work, but all this looks reasonable.

File: elftools/elf/relocation.py (hunk 1 above)
Comment (medhefgo) [reply elftools/elf/relocation.py:197]: It follows the same style already used by `RelocationSection`.

**Review by eliben — APPROVED** [review]

**Pull request merged**